concrete/src/Page/PageList.php

```diff
@@ -177,10 +177,13 @@ public function createQuery()
 
     public function finalizeQuery(\Doctrine\DBAL\Query\QueryBuilder $query)
     {
+        // avoid slow left join if we have an active page version in which case we always have a page path
+        $joinOperation = $this->pageVersionToRetrieve == self::PAGE_VERSION_ACTIVE ? 'innerJoin' : 'leftJoin';
```

Comment:
How about doing this after the if-else (line 204):

```
if(this->pageVersionToRetrieve == self::PAGE_VERSION_ACTIVE) {
$query->innerJoin('p', 'PagePaths', 'pp', 'p.cID = pp.cID and pp.ppIsCanonical = true')
}
else {
$query->leftJoin('p', 'PagePaths', 'pp', 'p.cID = pp.cID and pp.ppIsCanonical = true')
}
```

Reply:
Why would that fix the tests?